qcodes/dataset/sqlite/connection.py

```diff
@@ -151,3 +155,14 @@ def atomic_transaction(conn: ConnectionPlus,
     with atomic(conn) as atomic_conn:
         c = transaction(atomic_conn, sql, *args)
     return c
+
+
+def path_to_dbfile(conn: Union[ConnectionPlus, sqlite3.Connection]) -> str:
+    """
+    Return the path of the database file that the conn object is connected to
+    """
+    cursor = conn.cursor()
+    cursor.execute("PRAGMA database_list")
+    row = cursor.fetchall()[0]
+
+    return row[2]
```

Comment:
i liked the version of this function from an earlier commit that had more comments and links to sqlite docs or smth :)

Reply:
Yes that was from before I realized that we already had this function :) I can put the link here too

Reply:
This is the one that we always had. Just moved to make the dep tree work out